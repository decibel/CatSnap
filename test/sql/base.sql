\set ECHO none

\i test/pgxntool/setup.sql

SELECT plan(4);

SELECT is(
  (SELECT count(*)::int FROM _cat_snap.entity)
  , 101
  , 'Number of entities'
);

SELECT types_are(
  'cat_snap'
  , array( SELECT replace( entity, 'pg_', 'raw_' )::name FROM _cat_snap.entity )
    || array( SELECT replace( entity, 'pg_', 'delta_' )::name FROM _cat_snap.entity WHERE entity_type = 'Stats File' )
  , 'Verify types'
);

SELECT ok(
  (SELECT delta_keys @> array['queryid'] FROM _cat_snap.entity WHERE entity = 'pg_stat_statements')
  , 'Verify queryid exists in pg_stat_statements delta_keys'
);

SELECT is(
  (SELECT extra_attributes FROM _cat_snap.entity WHERE entity = 'pg_stat_activity')
  , array[row('waiting','boolean')::_cat_snap.attribute] 
  , 'Verify waiting exists in pg_stat_activity attributes'
);

\i test/pgxntool/finish.sql

-- vi: expandtab ts=2 sw=2
