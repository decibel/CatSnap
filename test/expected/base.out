\set ECHO none
1..14
ok 1 - Number of entities
ok 2 - Check expected missing entities
ok 3 - Verify types
ok 4 - Verify waiting exists in pg_stat_activity attributes
ok 5 - Verify gather code for 9.2 pg_stat_activity
ok 6 - Verify gather code for 9.3 pg_stat_activity
ok 7 - Verify gather code for 9.4 pg_stat_activity
ok 8 - Verify gather code for 9.5 pg_stat_activity
ok 9 - Verify gather code for 9.6 pg_stat_activity
ok 10 - SELECT cat_snap.snapshot_code('9.2', cluster_identifier := 'cluster id')
ok 11 - SELECT cat_snap.snapshot_code('9.3', cluster_identifier := 'cluster id')
ok 12 - SELECT cat_snap.snapshot_code('9.4', cluster_identifier := 'cluster id')
ok 13 - SELECT cat_snap.snapshot_code('9.5', cluster_identifier := 'cluster id')
ok 14 - SELECT cat_snap.snapshot_code('9.6', cluster_identifier := 'cluster id')
# TRANSACTION INTENTIONALLY LEFT OPEN!
